design/one-pager-considerations-for-changing-test-framework-of-uptest.md

```diff
@@ -0,0 +1,167 @@
+
+In addition to the functionality provided by `kuttl`, the prominent features of
+the `chainsaw` tool are that it is maintained up to date and has an
+infrastructure for migration from `kuttl`.
+
+There are very similar concepts in `chainsaw` with `kuttl`. By this way, we do
+not have to make important changes in the templates.
+
+```yaml
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: example
+spec:
+  steps:
+  - try:
+    # ...
+    - apply:
+        file: my-configmap.yaml
+    # ...
+```
+
+```yaml
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: example
+spec:
+  steps:
+  - try:
+    # ...
+    - command:
+        entrypoint: echo
+        args:
+        - hello chainsaw
+    # ...
+```
+
+Also provides logical assertion statements:
+
+```yaml
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: example
+spec:
+  steps:
+  - try:
+    # ...
+    - assert:
+        resource:
+          apiVersion: v1
+          kind: Deployment
+          metadata:
+            name: foo
+          spec:
+            (replicas > 3): true
+    # ...
+```
+
+Resource Template support is another important requirement for Uptest:
+
+```yaml
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: template
+spec:
+  template: true
+  steps:
+  - assert:
+      resource:
+        # apiVersion, kind, name, namespace and labels are considered for templating
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: ($namespace)
+        # other fields are not (they are part of the assertion tree)
+        data:
+          foo: ($namespace)
+```
+
+Related Issue: https://github.com/upbound/official-providers-ci/issues/179
+
+In short, the `chainsaw` is more capable and maintained tool than `kuttl`. So,
+switching to this new tool may better for handling the Uptest requirements.
+
+## Alternative Considered
+
+### Using the `crossplane-e2e-framework`
+
+[crossplane-e2e-framework](https://github.com/crossplane/crossplane/tree/master/test/e2e)
+is a [k8s-e2e-framework](https://pkg.go.dev/sigs.k8s.io/e2e-framework)-based
+test framework that provides a baseline for simulating the real-world use to
+exercise all of `Crossplane`'s functionality.
+
+`e2e-framework` is a tool that allows tests to be written in Go. Additionally,
+one of its advantages is that it works with familiar conventions in the
+environment we use. On the other hand, these types of utilities can be used when
+writing tests, thanks to their functions specific to the crossplane ecosystem.
+However, this will mean changing the entire `Uptest` code-base currently used.
+In this context, it should be taken into consideration that such a change would
+be quite large.
+
+Based on the discussion here, an analysis can be made to clarify the use cases
+of `Uptest` and the `e2e-framework` and which one the user would prefer
+according to their needs through documentation.
```

Comment:
I'm not following what is intended with this paragraph?

Reply:
I forgot to refer to the discussion. Added link. Also extended the expression.